lib/cloudregister/registerutils.py

```diff
@@ -1651,3 +1715,81 @@ def __replace_url_target(config_files, new_smt):
                     current_service_server,
                     new_smt.get_FQDN())
                 )
+
+
+# ----------------------------------------------------------------------------
+def _set_registry_order_search_podman(registry_fqdn):
+    public_registry_fqdn = 'registry.suse.com'
+    private_registry = {'location': registry_fqdn, 'insecure': False}
+    public_registry = {'location': public_registry_fqdn, 'insecure': False}
+    registries_conf = {}
+    try:
+        registries_conf = _get_registry_conf_file(
+            REGISTRIES_CONF_PATH, 'podman'
+        )
+        missing_public_registry = public_registry_fqdn not in \
+            registries_conf['unqualified-search-registries']
+        if missing_public_registry:
+            registries_conf['unqualified-search-registries'] = \
+                [public_registry_fqdn] + \
+                registries_conf['unqualified-search-registries']
+        if public_registry not in registries_conf['registry']:
+            registries_conf['registry'] = \
+                [public_registry] + registries_conf['registry']
+        if private_registry not in registries_conf['registry']:
+            registries_conf['registry'] = \
+                [private_registry] + registries_conf['registry']
+    except (FileNotFoundError, KeyError):
+        # file does not exist, create the file
+        os.makedirs(os.path.dirname(REGISTRIES_CONF_PATH), exist_ok=True)
+        registries_conf = _get_registry_conf_file(
+            REGISTRIES_CONF_PATH, 'podman'
+        )
+        # one or both keys do not exist
+        if registries_conf.get('unqualified-search-registries') is None:
+            registries_conf['unqualified-search-registries'] = \
+                ["{}".format(public_registry_fqdn)]
+        if registries_conf.get('registry') is None:
+            registries_conf['registry'] = \
+                [private_registry] + [public_registry]
+
+    with open(REGISTRIES_CONF_PATH, 'w') as registries_conf_file:
+        toml.dump(registries_conf, registries_conf_file)
```

Comment:
I think this part could be written a bit shorter and more clear with less duplication e.g 

```python
if not os.path.isfile(REGISTRIES_CONF_PATH):
    os.makedirs(os.path.dirname(REGISTRIES_CONF_PATH), exist_ok=True)

registries_conf = _get_registry_conf_file(
    REGISTRIES_CONF_PATH, 'podman'
)

registries_search: List[str] = registries_conf.get('unqualified-search-registries') or []
registries: List[str] = registries_conf.get('registry') or []

if public_registry_fqdn not in registries_search:      
    registries_search.append(public_registry_fqdn)

if public_registry not in registries:
    registries.append(public_registry)

if private_registry not in registries:
    registries.append(private_registry)

registries_conf['unqualified-search-registries'] = registries_search
registries_conf['registry'] = registries

with open(REGISTRIES_CONF_PATH, 'w') as registries_conf_file:
    toml.dump(registries_conf, registries_conf_file)
```

Reply:
and we are only adding the public registry to the search list, which I like because otherwise errors on authentication expiration will not be visible to the user, as we discussed it on the slack channel :+1:

Reply:
that would set the public registry fqdn at the list, I think the agreement was at the beginning ?

Reply:
ok so

```registries_search.insert(0, public_registry_fqdn)```

sorry I overlooked the order

Reply:
No worries

I took the same approach (insert o)